### ISSUE TITLE
Minor Bug: Converting from C to F by default 

### ISSUE THREAD

**Issue by tayvenb13** [andrewleech/ha_magiqtouch#25]
If your thermostat is set to Fahrenheit, and HomeAssistant is set to Fahrenheit; The sensor will take the temperature from the Thermostat as Celsius and then the UI will convert it to Fahrenheit. This results in the temperature showing absurdly high. 

The easy fix is to switch the Thermostat to Celsius. 

However I think the code itself could be changed to not [default as celsius](https://github.com/andrewleech/ha_magiqtouch/blob/6bfa8533f9e9bfa08508e0e8d4471033b350e590/custom_components/magiqtouch/climate.py#L199)  

I'd make a PR to help out, but to be honest I am not that great at coding. Just wanted to bring this up in case it happens to someone else :)

**Comment by e-pilot** [comment]
Been wondering what the heck was causing that, thanks for the tip

**Comment by andrewleech** [comment]
Oh yeah, thanks for the report. I'd wondered a bit myself what would happen if anyone wanted to use farenheit - it's not overly relevant down here in Australia :-D Shame it doesn't "just work" but not too surprising really.

The fix would involve reading from the magiqtouch whether it's set to C or F, but I don't know what that looks like! If you could change the config on the integration to turn on basic logging (like shown here https://github.com/andrewleech/ha_magiqtouch/issues/19#issuecomment-1987973747) then in the system logs there should be a couple of entries for magiqtouch with json dumps of the settings for your install. If you can copy them here I might be able to fix the issues.

**Comment by tayvenb13** [comment]
Of course! Apologies for the long response. Lost track of time getting this for you. 

`2024-07-06 14:53:40.423 WARNING (MainThread) [magiqtouch] Current System State: {"ACZones": {"Manual": true, "SlaveWallControls": false, "Zones": []}, "AOCFixed": {"InSystem": false, "MaximumTemperature": 70, "MinimumTemperature": 60}, "AOCInverter": {"MaximumTemperature": 70, "MinimumTemperature": 60, "InSystem": false}, "Cabinets": [{"CabinetSerialNo": 1132692, "CoolerCabinetSoftRev": "98R1120", "ElectronicsSerialNo": " ", "ModelNumber": "TBQI 5500"}], "EVAPCooler": {"Brands": 0, "HumidityControl": false, "MaximumTemperature": 70, "MinimumTemperature": 60, "TemperatureUnits": 1}, "WallController": {"Firmware": "44R0920", "Type": 0}, "Heater": {"InSystem": false, "FixedFan": false, "Brands": 0, "DataTableVersion": "D0.0", "ICSSoftwareRev": "D0.0", "MaxSetFanSpeed": 10, "MaximumTemperature": 100, "MinimumTemperature": 32, "ModelNo": "WIFISC", "SerialNo": 0}, "MasterAirSensorPresent": false, "SlaveWallControls": 0, "NoOfZoneControls": 0, "System": {"configuration": 0, "cooler": {"available": true, "fanFixed": false}, "heater": {"available": false, "fanFixed": false}, "fan": {"available": true, "fanFixed": false}, "Address": "<redacted>", "Name": "My Home"}, "Wifi_Module": {"MacAddressId": "<redacted>", "version": "1.9", "type": "WI-FI IoT Device"}, "ExternalAirSensorPresent": true, "DamperDelayModulePresent": false, "BMSS1": false, "BMSMS1": false, "ZoneAirSensors": 0}
2024-07-06 14:53:40.423 INFO (MainThread) [magiqtouch] Logged in`  

Then just in case its useful, here's the full log. 
[home-assistant_2024-07-06T20-55-53.095Z.log](https://github.com/user-attachments/files/16117347/home-assistant_2024-07-06T20-55-53.095Z.log)

**Comment by andrewleech** [comment]
Hi, I've tried to address this in https://github.com/andrewleech/ha_magiqtouch/commit/779548fd1e889ae6f111bcb95b78745e55fc7fd9
I'll get it in a release after a little more testing@

**Comment by andrewleech** [comment]
This was fixed a while back in the hacs branch (v5.0+), the integration now reads temperature_units from the device rather than hardcoding Celsius.
